### DISCUSSION TITLE
Microsoft child's account login issues

### DISCUSSION THREAD

**Discussion by hedgehog1029** [SKCraft/Launcher#397] [General]
It seems that 1.7.10 has some trouble creating a session when connecting to a server with Microsoft credentials; with the client displaying a "Failed to login: null" error message.

**Comment by hedgehog1029** [comment]
The error message indicates a generic, unknown error was returned by the session server, with no message attached (hence the displayed "null"). Ideally we could see what the non-blank "error" code is that triggers the exception to be thrown in the first place...

**Reply by sylphio** [reply]
Apparently, i got in sweet spot in term of bug ;p

Just tested with the minecraft launcher with a manual forge installation and i got the same problem.

You are right, it look server side for this last one.

**Reply by hedgehog1029** [reply]
From what I can tell, it's a mostly client-side issue - it errors when making a HTTP request to the session server (`sessionserver.mojang.com`) which it needs to do to check it's OK to continue the encryption dance with the server. Looking at the code, for the exception to be thrown at all the "error" field in the sessionserver response needs to be non-blank. So there might be a vaguely informative error code in that response.

**Reply by sylphio** [reply]
Okay, i got it... apparently, M$ have another level of parental control over that....

https://help.minecraft.net/hc/en-us/articles/360045816952-Additional-Parental-Controls-

I did that, and i was able to log for the first time.

I don't know if it is a fluke ;p

**Reply by sylphio** [reply]
It look fine now :) Thank

**Reply by hedgehog1029** [reply]
ah! that sounds like you've solved it. hopefully in 1.17 that error will be gracefully handled rather than a useless "null" error message, heh.

Or they could fix it on every version by just including a message with the error response. Hopefully they do that.

**Reply by sylphio** [reply]
It maybe helpful to add a link to the familly stuff + parental control somewhere in the wiki.

**Reply by sylphio** [reply]
I'm not be the only one with kid ;p

**Reply by hedgehog1029** [reply]
Yeah, when I get this merged in I'll be updating the wiki with lots of information about this. Apparently some of this doesn't happen when using the Minecraft client ID, but that Minecraft Help article seems to suggest the multiplayer problem *does* occur. At the very least the launcher should tell you now when the child account isn't added to a family (that was a bug where the error code wasn't being parsed properly)

**Reply by MadMartian** [reply]
Bump, I've got a small community of peeps (that I know of) actively (daily) using my 1.7.10-based mod.  Awesome work you're doing here, just merged your branch into our local, when I have some time later this week I want to take it for a spin.

**Reply by MadMartian** [reply]
> I'm not be the only one with kid ;p

I'm actually so confused, everyone knows Minecraft is a game for adults ;)
